### PR TITLE
[SYCLomatic] Add braces around kernel call that use root-group

### PR DESCRIPTION
**Pull request by yihwang-nv**

### CODE REVIEW

**Review by intwanghao — APPROVED** [review]
LGTM

File: clang/test/dpct/sync_api.cu [hunk 1]
```diff
@@ -106,12 +106,15 @@ __global__ void kernel() {
...
-  // CHECK:    auto exp_props = sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::use_root_sync};
+  // CHECK: {
+  // CHECK-NEXT:  auto exp_props = sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::use_root_sync};
+  // CHECK-EMPTY:
   // CHECK-NEXT:  dpct::get_in_order_queue().parallel_for(
   // CHECK-NEXT:  sycl::nd_range<3>(sycl::range<3>(1, 1, 2) * sycl::range<3>(1, 1, 2), sycl::range<3>(1, 1, 2)),
   // CHECK-NEXT:  exp_props,     [=](sycl::nd_item<3> item_ct1) {
   // CHECK-NEXT:      kernel(item_ct1);
   // CHECK-NEXT:    });
+  // CHECK-NEXT: }
```
Comment (zhimingwang36) [inline clang/test/dpct/sync_api.cu:117]: if there is only one kernel in the code block, then no need to insert {}.

File: clang/test/dpct/sync_api.cu [hunk 2]
```diff
@@ -106,12 +106,15 @@ __global__ void kernel() {
 }
 
 int main() {
-  // CHECK:    auto exp_props = sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::use_root_sync};
+  // CHECK: {
+  // CHECK-NEXT:  auto exp_props = sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::use_root_sync};
```
Comment (zhimingwang36) [inline clang/test/dpct/sync_api.cu:110]: can we inline the exp_props in parallel_for?

Comment (zhimingwang36) [inline clang/test/dpct/sync_api.cu:110]: How about insert the exp_pros when on demand? In this case, it is not required.

**Review by ziranzha — APPROVED** [review]

**Pull request merged**